pkg/gobmpsrv/gobmpsrv.go

```diff
@@ -68,7 +70,7 @@ func (srv *bmpServer) bmpWorker(client net.Conn) {
 		glog.V(5).Infof("connection to destination server %v established, start intercepting", server.RemoteAddr())
 	}
 	var producerQueue chan bmp.Message
-	prod := message.NewProducer(srv.publisher, srv.splitAF)
+	prod := message.NewProducer(srv.publisher, srv.adminId, srv.splitAF)
```

Comment:
Since admin ID is somewhat publisher requirement, it is better to keep in in a publisher. You can add a method to publisher something like getAdminId and call it when message gets published adding it to the produced message.